language/vm/vm_runtime/src/data_cache.rs

```diff
@@ -265,8 +269,15 @@ impl<'txn> TransactionDataCache<'txn> {
         }
 
         // Insert the code blob to the writeset.
-        for (key, blob) in to_be_published_modules.into_iter() {
-            write_set.push(((&key).into(), WriteOp::Value(blob)));
+        if write_set.len() <= usize::max_value() - to_be_published_modules.len() {
+            for (key, blob) in to_be_published_modules.into_iter() {
+                write_set.push(((&key).into(), WriteOp::Value(blob)));
+            }
+        } else {
+            return Err(VMRuntimeError {
```

Comment:
Is this the right kind of error to return if the write set grows out of bound? How about adding another type of VM Error?

Reply:
You're right; I think that `VMErrorKind::InvalidData` is actually more appropriate to express this error.